docs/content/utilities/validation.mdx

```diff
@@ -0,0 +1,236 @@
+
+You can use our built-in JMESPath functions within your expressions to do exactly that to decode JSON Strings, base64, and uncompress gzip data.
+
+<Note type="info">
+    We use these for built-in envelopes to easily to decode and unwrap events from sources like Kinesis, CloudWatch Logs, etc.
+</Note>
+
+### powertools_json function
+
+Use `powertools_json` function to decode any JSON String.
+
+This sample will decode the value within the `data` key into a valid JSON before we can validate it.
+
+```python:title=powertools_json_jmespath_function.py
+from aws_lambda_powertools.utilities.validation import validate
+
+json_schema_dict = {..}
+sample_event = {
+    'data': '{"payload": {"message": "hello hello", "username": "blah blah"}}'
+}
+
+def handler(event, context):
+	validate(event=event, schema=json_schema_dict, envelope="powertools_json(data)") # highlight-line
+	return event
+
+handler(event=sample_event, context={})
+```
+
+### powertools_base64 function
+
+Use `powertools_base64` function to decode any base64 data.
+
+This sample will decode the base64 value within the `data` key, and decode the JSON string into a valid JSON before we can validate it.
+
+```python:title=powertools_json_jmespath_function.py
+from aws_lambda_powertools.utilities.validation import validate
+
+json_schema_dict = {..}
+sample_event = {
+    "data": "eyJtZXNzYWdlIjogImhlbGxvIGhlbGxvIiwgInVzZXJuYW1lIjogImJsYWggYmxhaCJ9="
+}
+
+def handler(event, context):
+	validate(event=event, schema=json_schema_dict, envelope="powertools_json(powertools_base64(data))") # highlight-line
+	return event
+
+handler(event=sample_event, context={})
+```
+
+### powertools_base64_gzip function
+
+Use `powertools_base64_gzip` function to decompress and decode base64 data.
+
+This sample will decompress and decode base64 data, then use JMESPath pipeline expression to pass the result for decoding its JSON string.
+
+```python:title=powertools_json_jmespath_function.py
+from aws_lambda_powertools.utilities.validation import validate
+
+json_schema_dict = {..}
+sample_event = {
+    "data": "H4sIACZAXl8C/52PzUrEMBhFX2UILpX8tPbHXWHqIOiq3Q1F0ubrWEiakqTWofTdTYYB0YWL2d5zvnuTFellBIOedoiyKH5M0iwnlKH7HZL6dDB6ngLDfLFYctUKjie9gHFaS/sAX1xNEq525QxwFXRGGMEkx4Th491rUZdV3YiIZ6Ljfd+lfSyAtZloacQgAkqSJCGhxM6t7cwwuUGPz4N0YKyvO6I9WDeMPMSo8Z4Ca/kJ6vMEYW5f1MX7W1lVxaG8vqX8hNFdjlc0iCBBSF4ERT/3Pl7RbMGMXF2KZMh/C+gDpNS7RRsp0OaRGzx0/t8e0jgmcczyLCWEePhni/23JWalzjdu0a3ZvgEaNLXeugEAAA=="
+}
+
+def handler(event, context):
+	validate(event=event, schema=json_schema_dict, envelope="powertools_base64_gzip(data) | powertools_json(@)") # highlight-line
+	return event
+
+handler(event=sample_event, context={})
+```
+
+## Bring your own JMESPath function
+
+<Note type="warning">
+    This should only be used for advanced use cases where you have special formats not covered by the built-in functions.
+    <br/><br/>
+    This will <strong>replace all built-in functions provided with what you provide</strong>.
```

Comment:
```suggestion
    This will <strong>replace all provided built-in functions such as `powertools_json`, so you will no longer be able to use them</strong>.
```